main.cc

```diff
@@ -1356,7 +1356,10 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             service::raft_group0 group0_service{
                     stop_signal.as_local_abort_source(), raft_gr.local(), messaging.local(),
-                    gossiper.local(), qp.local(), mm.local(), group0_client};
+                    gossiper.local(), qp.local(), mm.local(), group0_client,
+                    proxy.local().get_db().local().commitlog()
+                        ? proxy.local().get_db().local().commitlog()->max_record_size()
+                        : 0};
```

Comment:
Is this max_record_size directly translatable? That is, there's no extra padding or signatures that commitlog adds on top, which then mean we have an off-by-small-constant error here?

There's another problem in that different replicas may have different max record size configuration, but dealing with that is much harder (plus it never happens).

Reply:
`max_record_size` already [includes](https://github.com/scylladb/scylladb/blob/master/db/commitlog/commitlog.cc#L2531) some paddings, and [the test](https://github.com/scylladb/scylladb/blob/master/test/boost/commitlog_test.cc#L265) implies that `max_record_size` could be treated as a public declaration of the limit (the test lacks the success scenario though).

However, a raft command can include many mutations, [num_entries](https://github.com/scylladb/scylladb/blob/master/db/commitlog/commitlog.cc#L180) will be [equal](https://github.com/scylladb/scylladb/blob/master/db/commitlog/commitlog.cc#L2405) to the number of mutations, and we [can get](https://github.com/scylladb/scylladb/blob/master/db/commitlog/commitlog.cc#L1139) a bigger size eventually. I'll handle it and write a separate test.

Reply:
IIRC when we persist mutations we write each one individually into a commit log (what we actually do it an unlogged batch write).

Reply:
What if the raft overhead changes over time? An old node makes a decision here for a combination of old and new nodes.

We need drastic safety margins here, like dividing by two.

Reply:
And also, if/when we lift commitlog restrictions, we must do it in a coordinated way so mixed clusters work.

Reply:
added division by two